enterprise/cli/server_dbcrypt_test.go

```diff
@@ -191,30 +216,84 @@ func TestServerDBCrypt(t *testing.T) {
 	}
 }
 
-func genData(t *testing.T, db database.Store, n int) []database.User {
+func genData(t *testing.T, db database.Store) []database.User {
 	t.Helper()
 	var users []database.User
-	for i := 0; i < n; i++ {
-		usr := dbgen.User(t, db, database.User{
-			LoginType: database.LoginTypeOIDC,
-		})
-		_ = dbgen.UserLink(t, db, database.UserLink{
-			UserID:            usr.ID,
-			LoginType:         usr.LoginType,
-			OAuthAccessToken:  "access-" + usr.ID.String(),
-			OAuthRefreshToken: "refresh-" + usr.ID.String(),
-		})
-		_ = dbgen.GitAuthLink(t, db, database.GitAuthLink{
-			UserID:            usr.ID,
-			ProviderID:        "fake",
-			OAuthAccessToken:  "access-" + usr.ID.String(),
-			OAuthRefreshToken: "refresh-" + usr.ID.String(),
-		})
-		users = append(users, usr)
+	// Make some users
+	for _, status := range database.AllUserStatusValues() {
+		for _, loginType := range database.AllLoginTypeValues() {
+			for _, deleted := range []bool{false, true} {
+				usr := dbgen.User(t, db, database.User{
+					LoginType: loginType,
+					Status:    status,
+					Deleted:   deleted,
+				})
+				_ = dbgen.GitAuthLink(t, db, database.GitAuthLink{
+					UserID:            usr.ID,
+					ProviderID:        "fake",
+					OAuthAccessToken:  "access-" + usr.ID.String(),
+					OAuthRefreshToken: "refresh-" + usr.ID.String(),
+				})
+				// Fun fact: our schema allows _all_ login types to have
+				// a user_link. Even though I'm not sure how it could occur
+				// in practice, making sure to test all combinations here.
+				_ = dbgen.UserLink(t, db, database.UserLink{
+					UserID:            usr.ID,
+					LoginType:         usr.LoginType,
+					OAuthAccessToken:  "access-" + usr.ID.String(),
+					OAuthRefreshToken: "refresh-" + usr.ID.String(),
+				})
+				users = append(users, usr)
+			}
+		}
 	}
 	return users
 }
 
+func dumpUsers(t *testing.T, db *sql.DB) {
+	t.Helper()
+	rows, err := db.QueryContext(context.Background(), `SELECT
+	u.id,
+	u.login_type,
+	u.status,
+	u.deleted,
+	ul.oauth_access_token_key_id AS uloatkid,
+	ul.oauth_refresh_token_key_id AS ulortkid,
+	gal.oauth_access_token_key_id AS galoatkid,
+	gal.oauth_refresh_token_key_id AS galortkid
+FROM users u
+LEFT OUTER JOIN user_links ul ON u.id = ul.user_id
+LEFT OUTER JOIN git_auth_links gal ON u.id = gal.user_id
+ORDER BY u.created_at ASC;`)
```

Comment:
Maybe we should add a test that this function works as expected since it's static and will only be run on failure, meaning it will most likely be outdated by the time it's needed. 😆

Perhaps this could be motivation for a `testqueries` package. Similar to what I did here: https://github.com/coder/coder/pull/9519 to generate a new (slim) package that can be wrapped around `sql.DB`.

Reply:
I'd definitely be in favour of this; I'm fairly sure there are at least a couple of queries that only ever get run in unit tests.

Reply:
> Maybe we should add a test that this function works as expected since it's static and will only be run on failure, meaning it will most likely be outdated by the time it's needed. 😆

Golden files?

Reply:
I think verifying a few fields in the test is enough. Otherwise you need to stabilize the generated IDs.

Reply:
I think instead of making that query more complex I'll go ahead and add a follow-up PR to dump the entire test database on failure (optional, opt-in). Less work, less fuss, more benefit.